lib/vmpooler/api/v2.rb

```diff
@@ -24,6 +65,56 @@ def get_template_aliases(template)
         end
       end
 
+      def get_pool_weights(template_backends)
+        pool_index = pool_index(pools)
+        weighted_pools = {}
+        template_backends.each do |t|
+          next unless pool_index.key? t
+  
+          index = pool_index[t]
+          clone_target = pools[index]['clone_target'] || config['clone_target']
+          next unless config.key?('backend_weight')
+  
+          weight = config['backend_weight'][clone_target]
+          if weight
+            weighted_pools[t] = weight
+          end
+        end
+        weighted_pools
+      end
+  
+      def count_selection(selection)
+        result = {}
+        selection.uniq.each do |poolname|
+          result[poolname] = selection.count(poolname)
+        end
+        result
+      end
+  
+      def evaluate_template_aliases(template, count)
+        template_backends = []
+        template_backends << template if backend.sismember('vmpooler__pools', template)
+        selection = []
+        aliases = get_template_aliases(template)
+        if aliases
+          template_backends += aliases
+          weighted_pools = get_pool_weights(template_backends)
+  
+          if weighted_pools.count > 1 && weighted_pools.count == template_backends.count
+            pickup = Pickup.new(weighted_pools)
+            count.to_i.times do
+              selection << pickup.pick
+            end
+          else
+            count.to_i.times do
+              selection << template_backends.sample
+            end
+          end
+        end
```

Comment:
Better form is to avoid multiple if/else layers using guardian clauses. This increases code maintainability by reducing code paths. Here's an example:

```ruby
def evaluate_template_aliases(template, count)
  [ ... ]
  aliases =  get_template_aliases(template)
  return [] unless aliases

  selection = []
  template_backends += aliases
  weighted_pools = get_pool_weights(template_backends)

  if weighted_pools.count > 1 && weighted_pools.count == template_backends.count
    pickup = Pickup.new(weighted_pools)
    count.to_i.times do
      selection << pickup.pick
    end
    return selection
  end

  count.to_i.times do
    selection << template_backends.sample
  end

  return selection
end

Reply:
For all the comments in this file, everything here was just a straight copy/paste from the apiv1 to apiv2, since previously this v2 inherited from v1.